cmd/utils.go

```diff
@@ -301,13 +363,46 @@ func getOperatorHome(config *RootCommandConfig) string {
 }
 
 func getProjectName(config *RootCommandConfig) (string, error) {
-	projectDir, err := getProjectDir(config)
+	dir, err := getProjectDir(config)
 	if err != nil {
 		return "my-project", err
 	}
-	projectName := strings.ToLower(filepath.Base(projectDir))
-	projectName = strings.ReplaceAll(projectName, "_", "-")
-	return projectName, nil
+
+	appsodyConfig := filepath.Join(dir, ConfigFile)
+	v := viper.New()
+	v.SetConfigFile(appsodyConfig)
+	err = v.ReadInConfig()
+
+	match, _ := regexp.MatchString("[a-z]([-a-z0-9]*[a-z0-9])?", v.GetString("project-name"))
```

Comment:
Still need to error out if the `project-name` exists, but does not match the regex.

Reply:
changed